lib/MochaWatch.js

```diff
@@ -64,15 +69,7 @@ module.exports = class MochaWatch extends EventEmitter {
     this.debug("added", path);
     path = resolve(this.cwd, path);
 
-    const testFiles = await findFilesMochaWouldRun(this.program);
-
-    // If the added file is a test file we need to repopulate the sourceGraph,
-    // otherwise we can just add the file.
-    if (testFiles.includes(path)) {
-      await this.sourceGraph.populate(testFiles);
-    } else {
-      await this.sourceGraph.addFileFromPath(path);
-    }
+    this.queueFile(path);
```

Comment:
It would make for a more elegant API if the paths could be passed to the `queueTestRun` method, or if `queueFile` would call `queueTestRun` internally.